Zend/zend_smart_str.c

```diff
@@ -52,7 +55,10 @@ ZEND_API void ZEND_FASTCALL smart_str_realloc(smart_str *str, size_t len)
 		ZSTR_LEN(str->s) = 0;
 	} else {
 		str->a = SMART_STR_NEW_LEN(len);
-		str->s = (zend_string *) realloc(str->s, str->a + _ZSTR_HEADER_SIZE + 1);
+		void *temp = realloc(str->s, str->a + _ZSTR_HEADER_SIZE + 1);
```

Comment:
We can't handle this failure gracefully, so this should use the infallible `pemalloc(..., ..., 1)` instead.